activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1135,6 +1135,8 @@ def build_where_clause(opts, rest = []) # :nodoc:
           opts = opts.transform_keys do |key|
             key = key.to_s
             klass.attribute_aliases[key] || key
+          end.transform_values do |value|
+            value.class == Enumerator::Lazy ? value.to_a : value
           end
```

Comment:
Not sure (yet) if maintainers accept this behavior change, it probably need to adapt the way it re-assigns the `opts` local variable. Instead of create two hashes on the double iteration, use `each_with_object`:

```suggestion
          opts = opts.each_with_object do |(key, value), hash|
            key = key.to_s
            hash[klass.attribute_aliases[key] || key] = value.is_a?(Enumerator::Lazy) ? value.to_a : value
          end
```